core/staking_verifier.go

```diff
@@ -55,9 +55,10 @@ func VerifyAndCreateValidatorFromMsg(
 	wrapper.Delegations = []staking.Delegation{
 		staking.NewDelegation(v.Address, msg.Amount),
 	}
-	wrapper.Snapshot.Epoch = epoch
-	wrapper.Snapshot.NumBlocksSigned = big.NewInt(0)
-	wrapper.Snapshot.NumBlocksToSign = big.NewInt(0)
+	zero := big.NewInt(0)
+	wrapper.LastEpochInCommittee = epoch
```

Comment:
don't change LastEpochInCommittee. It's wrong here. LastEpochInCommittee is 0 if the validator is never selected in committee (new validator)